[Ng Jia Yi] iP

gremmyz opened the pull request:
## DukePro
>"Your mind is for having ideas, not holding them." – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://github.com/gremmyz/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
---
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

gremmyz committed:
Add echo

gremmyz committed:
add Level-2 of Duke

gremmyz committed:
updated Duke Task-2

gremmyz committed:
added checkbox for Level-3

gremmyz committed:
added Level-3

gremmyz committed:
add Deadline, Event, Todo class

gremmyz committed:
add task types for Level-4

gremmyz committed:
add TextUiTesting

gremmyz committed:
Update Duke.java

gremmyz committed:
add Exceptions

gremmyz committed:
Update Duke.java

gremmyz committed:
add Enums

gremmyz committed:
update DukeBot

gremmyz committed:
update Level-7: Save

gremmyz committed:
update Level-8

gremmyz committed:
add Ui, Storage, Parser, TaskList.

gremmyz committed:
Refactor classes, add into Duke package.

gremmyz committed:
Merge remote-tracking branch 'origin/add-gradle-support'

gremmyz committed:
add Gradle support

gremmyz committed:
Add JUnit Tests for Todo, Deadline and Event.

gremmyz committed:
Add package as JAR file function.

gremmyz committed:
Add JavaDoc.

gremmyz committed:
Merge branch 'branch-A-JavaDoc'

gremmyz committed:
Add CodingStandard

gremmyz committed:
Merge branch 'branch-A-CodingStandard'

gremmyz committed:
Add Level-9: Find.

gremmyz committed:
Merge branch 'branch-Level-9'

gremmyz committed:
Update README.md

gremmyz committed:
Add A-CheckStyle.

gremmyz committed:
Merge branch 'branch-A-CheckStyle'

evanpy commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,16 @@
+public class Event extends Task {
+
+    protected String from;
+    protected String to;
```
"from" and "to" might be ambiguous, well at least from my pov. Instead of time, readers could think of it in terms of place.
Other than that, nice code 👍🏻 good to go.

gremmyz replied:
Thanks for the feedback! You're right, will adjust as necessary 👍🏼

evanpy reviewed (commented):
LGTM! 😄

gremmyz committed:
Add Level-10: GUI.

gremmyz committed:
Merge branch 'branch-Level-10'

gremmyz committed:
Add Level-10: GUI.

junianob commented on lists/taskList.txt:
```diff
@@ -0,0 +1,3 @@
+T | 0 | yeet
+T | 0 | eat some bread
+E | 0 | pet some cats | 2023-12-02
```
Maybe you might want to keep your taskList private?
You can consider adding lists/taskList.txt to your .gitignore file!

junianob commented on src/main/java/seedu/duke/Event.java:
```diff
@@ -0,0 +1,76 @@
+     *
+     * @return String for Event creation
+     */
+    public static String runEvent(TaskList taskList, String description) {
```
You used description for the variable name here and desc somewhere else, would it be preferable to stick to a consistent naming scheme?

gremmyz replied:
Thanks for the feedback! I agree with you, will make the necessary amendments.

junianob commented on src/main/java/seedu/duke/Todo.java:
```diff
@@ -0,0 +1,64 @@
+     *
+     * @return String for Todo creation.
+     */
+    public static String createTodo(TaskList taskList, String desc) {
```
You used desc for the variable name here and description somewhere else, would it be preferable to stick to a consistent naming scheme?

junianob commented on src/main/java/seedu/duke/Parser.java:
```diff
@@ -0,0 +1,115 @@
+            case BYE:
+                response = ui.displayExit();
+                break;
+                // Command for list
```
not inline! Maybe wrote the comment on the wrong line?

gremmyz replied:
Fixed it!

junianob commented on src/main/java/seedu/duke/Ui.java:
```diff
@@ -0,0 +1,159 @@
+                list = list + "\n   " + index + ". " + taskList.get(index - 1).toString();
+                index++;
+            }
+            String hasTasks = "Here are your tasks:";
```
Would it be cleaner to return in a single statement without the hasTasks String?

gremmyz replied:
Amended!

junianob commented on src/main/java/seedu/duke/Ui.java:
```diff
@@ -0,0 +1,159 @@
+                list = list + "\n   " + index + ". " + taskList.get(index - 1).toString();
+                index++;
+            }
+            String hasTasks = "Here are the tasks with the keyword: ";
```
Would it be cleaner to return in a single statement without the hasTasks String?

gremmyz replied:
I agree with you, my original intention was so that the return statement would be in a single line for me to read it easily 😄

junianob reviewed (commented):
LGTM! Coding standard is followed and class, method and fields are named appropriately. Minor nitpicks/suggestions that can be trivially handled. :)

gremmyz committed:
Update .gitignore.

gremmyz committed:
Updated files based on feedback.

gremmyz committed:
Added A-Assertions.

gremmyz committed:
Merge branch 'master' into branch-A-Assertions

gremmyz committed:
Merge pull request #1 from gremmyz/branch-A-Assertions

Added A-Assertions.

gremmyz committed:
Added A-Assertions.

Add assertions to Ui and TaskList methods.

There are several methods in Ui and TaskList that cannot be negative. The TaskList should not be negative, as it will indicate an error otherwise and terminate the program. It is better to assert this instead of terminating the program and having the user run it again.

gremmyz committed:
Update A-CodeQuality.

gremmyz committed:
Merge branch 'master' into branch-A-CodeQuality

gremmyz committed:
Merge pull request #2 from gremmyz/branch-A-CodeQuality

Some classes and methods have incorrect Java coding standard.

By fixing the code quality, the code becomes more readable. Other users
and developers will be able to understand the code easier.

Let's
* review feedback to coding standard and code quality from PR reviews.
* standardize naming conventions of variables (in particular, desc and description).
* update return statement to make code cleaner.

gremmyz committed:
Add A-CI.

Setting up a custom workflow.

gremmyz committed:
Add extension C-Help.

gremmyz committed:
Merge branch 'C-Help'

gremmyz committed:
Update extension C-Help.

gremmyz committed:
Add Ui.png

gremmyz committed:
Add Ui.png

gremmyz committed:
Update build.gradle

Supports GUI.

gremmyz committed:
Update A-UserGuide to include newest release.

gremmyz committed:
Update methods from code quality feedback.

gremmyz committed:
Update README.md
